### ISSUE TITLE
[rllib] WandbLoggerCallback crashes if a gym.spaces.Dict object is in the config dictionary. 

### ISSUE THREAD

**Issue by mgerstgrasser** [ray-project/ray#16681]
### What is the problem?
WandbLoggerCallback fails if there is a gym.spaces.Dict object in the rllib config dictionary, e.g. because an agent's observation space is being set to this. The problem seems to be that [wandb.py:34](https://github.com/ray-project/ray/blob/master/python/ray/tune/integration/wandb.py#L34) tries to access `len()` of the gym.spaces.Dict object, which fails. 

```
  File "[...]/main.py", line 21, in <module>
    log_config=True,
  File "[...]/lib/python3.7/site-packages/ray/tune/tune.py", line 520, in run
    runner.step()
  File "[...]/lib/python3.7/site-packages/ray/tune/trial_runner.py", line 501, in step
    if _start_trial(next_trial):
  File "[...]/lib/python3.7/site-packages/ray/tune/trial_runner.py", line 495, in _start_trial
    trial=trial)
  File "[...]/lib/python3.7/site-packages/ray/tune/callback.py", line 192, in on_trial_start
    callback.on_trial_start(**info)
  File "[...]/lib/python3.7/site-packages/ray/tune/logger.py", line 397, in on_trial_start
    self.log_trial_start(trial)
  File "[...]/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 344, in log_trial_start
    config = _clean_log(config)
  File "[...]/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 42, in _clean_log
    return {k: _clean_log(v) for k, v in obj.items()}
  File "[...]/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 42, in <dictcomp>
    return {k: _clean_log(v) for k, v in obj.items()}
  File "[...]/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 45, in _clean_log
    elif _is_allowed_type(obj):
  File "[...]/lib/python3.7/site-packages/ray/tune/integration/wandb.py", line 34, in _is_allowed_type
    if isinstance(obj, Iterable) and len(obj) > 0:
TypeError: object of type 'Dict' has no len()
```

*Ray version and other system information (Python version, TensorFlow version, OS):*
Tested in 1.4.0 and 2.0.0dev0

### Reproduction (REQUIRED)

```
import os
import ray
import ray.rllib.agents.a3c.a2c as a2c
from ray import tune
from ray.tune.integration.wandb import WandbLoggerCallback
import gym


ray.init()
config = a2c.A2C_DEFAULT_CONFIG.copy()
obs_space = gym.spaces.Dict({"key1": gym.spaces.Discrete(2)})
config["something"] = obs_space

tune.run(
    a2c.A2CTrainer,
    config=config,
    callbacks=[
        WandbLoggerCallback(
            project="test",
            api_key=os.environ["WANDB_API_KEY"],
            log_config=True,
        )
    ],
)
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by krfricke** [comment]
Thanks! I fixed this in #16806.

Further I filed https://github.com/openai/gym/pull/2249 which will make `Dict` spaces proper `Mapping`s

**Comment by mgerstgrasser** [comment]
Brilliant, thank you! I can confirm that #16806 solves the issue for us.
